worker/execute/session.py

```diff
@@ -0,0 +1,20 @@
+from util.network import get_local_ip, get_random_port
+
+
+class Session:
+    """
+    Base class for alternative types of job sessions.
+    """
+
+    protocol: str
+    ip: str = get_local_ip()
+    port: int
+
+    def __init__(self, protocol: str = "ws"):
+        self.protocol = protocol
+        self.ip = Session.ip
```

Comment:
This is a bit strange because in this case `self.ip` already equals `Session.ip` so this line is redundant. Unless you've updated a property on `self` it will be equivalent to its class value.